### ISSUE TITLE
Packer Naming conventions similar to Terraform

### ISSUE THREAD

**Issue by AzySir** [hashicorp/packer#11393]
I'd like to make a request, now that Packer uses HCL2 is it possible to start following a more consistent approach with how terraform file naming is done. 

I can't comprehend why in the instance of terraform vars we have `<name>.tfvars` but for packer we have `<name>.pkrvars.hcl` 

The curiosity was killing me so I thought I have to ask. Surely the vars could just be a pkvars and/or a .hcl and not both? It's too wordy for a file extension. 

Furthermore I have also noticed the pkr.hcl when I google it seems there is a file extension that uses .pkr - then can't we change this to anything else that does not require a .hcl ending? .pcr/.pckr/.pk/ etc.

**Comment by azr** [comment]
Hello @AzySir, good question !

The `.tfvars` ext is kind of a legacy extension. It has been here for a while, but they are `hcl` files and in order to not create a new file extension each time a new (HashiCorp or not) project adopts hcl; going forward all hcl files will get an `.hcl` extension, so that editors, GitHub, etc. can pick up that they are in fact hcl files without us having to 'say so'.

Also, it would be nice if terraform var files got a new hcl extension, but given the impact that this would have, this probably most likely will never change (somone in the tf team would be better fitted to answer this question) :)

**Comment by azr** [comment]
I'm going to go ahead a close this issue, but feel free to comment etc. Thanks !

**Comment by azr** [comment]
Good question too ! So, HCL is not exactly like JSON in the sense that each Tool has its own schema, some are slightly similar, like Packer and Terraform but, still with differences and some a super dissimilar, like say Packer and Consul.

Having the possibility to differentiate between those makes it easier for a code editor and other user tools to know what kind of file it is. This could enable validation, autocompletion & etc. But in the end, a Terraform schema is not compatible with a Packer schema, and we needed a simple way to pick that up. All whilst at the same time a simple way to tell that it's an HCL file. Another good thing is that a Terraform run will never try to read HCL files and vice versa.

> It's a little unorthodox and gets really wordy.

I agree ! 🙂  It's a tradeoff choice we made some time ago, and I still think it's generally a good option.

> Is there a .auto.hcl equivalent to .auto.tfvars in terraform?

You can create `*.auto.pkrvars.hcl` files in the build files folder. More in the [docs](https://www.packer.io/guides/hcl/variables)

**Comment by AzySir** [comment]
Thanks again for your reply @azr  i'll keep these in mind. 

That piece of work is finished. I can't say I'm a fan of this change file naming conventions. I do understand it.

Terraform is nice in the sense that you know that a `.tfvars` file contains you vars. 

IF they're all .hcl files have to click through all of them. Ofcourse you could call it variables.hcl but yeah... anyway - thank you again!
